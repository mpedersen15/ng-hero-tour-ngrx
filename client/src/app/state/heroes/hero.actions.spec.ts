import { LoadAll, HeroActionTypes, LoadAllSuccess } from './hero.actions';
import { Hero } from '../../core/models/hero.model';

describe('Hero LoadAll', () => {
    test('should have type equal to LoadAll', () => {
        const action = new LoadAll();

        expect(action.type).toBe(HeroActionTypes.LoadAll);
    });

    test('LoadAllSuccess should have heroes payload', () => {
        const heroes: Hero[] = [
            {
                id: 1,
                name: 'Thor',
                powers: ['Lightning'],
                character: {
                    id: 1,
                    name: 'Thor',
                    description: '',
                    urls: [''],
                    thumbnail: ''
                }
            }
        ]
        const action = new LoadAllSuccess({ heroes });

        // expect(action.type).toBe(HeroActionTypes.LoadAllSuccess);
        // expect(action.payload.heroes).toBe(heroes);

        expect(action.type).toMatchSnapshot();
        expect(action.payload).toMatchSnapshot();
    });
});


