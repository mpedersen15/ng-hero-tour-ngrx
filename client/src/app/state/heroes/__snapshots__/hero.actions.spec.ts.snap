// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Hero LoadAll LoadAllSuccess should have heroes payload 1`] = `"[Hero] Load Heroes Success"`;

exports[`Hero LoadAll LoadAllSuccess should have heroes payload 2`] = `
Object {
  "heroes": Array [
    Object {
      "character": Object {
        "description": "",
        "id": 1,
        "name": "Thor",
        "thumbnail": "",
        "urls": Array [
          "",
        ],
      },
      "id": 1,
      "name": "Thor",
      "powers": Array [
        "Lightning",
      ],
    },
  ],
}
`;
